### PR TITLE
Reworded the article about form login redirects

### PR DESCRIPTION
**Pull request by javiereguiluz**
Now that form login redirects have been fully fixed (see https://github.com/symfony/symfony/pull/23580) I thought about updating this article, specially its structure.

All changes are simple rewordings, except this one: previously, the article said that you can use a Symfony route name as the value of the `_target_path` parameter in the query string or the hidden form field. But if you check the code of this feature, it looks like you can't because we use the value of that parameter "as is" to redirect, so it must be a relative/absolute URL, right?

```php
protected function determineTargetUrl(Request $request)
{
    if ($this->options['always_use_default_target_path']) {
        return $this->options['default_target_path'];
    }

    // We redirect directly to the value of the parameter, so it can't be a route name, right ????
    if ($targetUrl = $request->get($this->options['target_path_parameter'], null, true)) {
        return $targetUrl;
    }

    // ...
}
```

### CODE REVIEW

**Comment by yceruto** [comment]
Please, could you take account this comment https://github.com/symfony/symfony/issues/17529#issuecomment-316390347 in this PR?

**Comment by yceruto** [comment]
See @dmaicher's [comment](https://github.com/symfony/symfony/issues/17529#issuecomment-204534951) which also explains the only use case of this option:
> So it only works for use cases where the form POST comes from a different page than /login
> Maybe its meant to be used if you have the login form embedded in the header of your page or something. So once you logged in you get redirected back to the page where the POST came from.

**Comment by javiereguiluz** [comment]
@yceruto thanks for letting me know this. I've updated the article. Cheers!

File: security/form_login.rst [hunk 1]
```diff
@@ -1,50 +1,42 @@
...
-Form Login Configuration Reference
-----------------------------------
-
-To see the full form login configuration reference, see
-:doc:`/reference/configuration/security`. Some of the more interesting options
-are explained below.
+Using a :doc:`form login </security/form_login_setup>` for authentication is a
+common, and flexible, method for handling authentication in Symfony. This
+article explains how to customize the URL which the user is redirected to after
+a successful or failure login. Check out the full
```
Comment (xabbuh) [inline security/form_login.rst]: "[...] or failed login."?

File: security/form_login.rst [hunk 2]
```diff
@@ -1,50 +1,42 @@
...
+:doc:`form login configuration reference </reference/configuration/security>` to
+learn about the rest of possible customizations.
```
Comment (xabbuh) [inline security/form_login.rst]: [...] of the possile customization options.

File: security/form_login.rst [hunk 3]
```diff
@@ -1,50 +1,42 @@
...
-then after they successfully log in, they will eventually be sent back to
-``http://www.example.com/admin/post/18/edit``.
-This is done by storing the requested URL in the session.
-If no URL is present in the session (perhaps the user went
-directly to the login page), then the user is redirected to the default page,
-which is  ``/`` (i.e. the homepage) by default. You can change this behavior
-in several ways.
+By default, the form will redirect to the URL the user requested (i.e. the URL
+which triggered the login form being shown). For example, if the user requested
+``http://www.example.com/admin/post/18/edit``, then after they successfully log
```
Comment (xabbuh) [inline security/form_login.rst]: [...] they have successfully logged in, [...]

File: security/form_login.rst [hunk 4]
```diff
@@ -1,50 +1,42 @@
...
 
 Changing the default Page
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-First, the default page can be set (i.e. the page the user is redirected to
-if no previous page was stored in the session). To set it to the
-``default_security_target`` route use the following config:
+Define the ``default_security_target`` option to change the page where the user
+is redirected to if no previous page was stored in the session. The value can be
+ relative/absolute URL or a Symfony route name:
```
Comment (xabbuh) [inline security/form_login.rst]: [...] can be a relative/absolute [...]

File: security/form_login.rst [hunk 5]
```diff
@@ -159,12 +147,52 @@ of what URL they had requested previously by setting the
...
+
+    http://example.com/some/path?_target_path=/dashboard
+
+Defining the redirect URL via POST using a hidden form field:
+
+.. configuration-block::
+
+    .. code-block:: html+twig
+
+        {# src/AppBundle/Resources/views/Security/login.html.twig #}
```
Comment (xabbuh) [inline security/form_login.rst]: `{# app/Resources/views/security/login.html.twig #}`

File: security/form_login.rst [hunk 6]
```diff
@@ -159,12 +147,52 @@ of what URL they had requested previously by setting the
...
+        <form action="{{ path('login') }}" method="post">
+            {# ... #}
+
+            <input type="hidden" name="_target_path" value="{{ path('account') }}" />
+            <input type="submit" name="login" />
+        </form>
+
+    .. code-block:: html+php
+
+        <!-- src/AppBundle/Resources/views/Security/login.html.php -->
```
Comment (xabbuh) [inline security/form_login.rst]: `<!-- app/Resources/views/security/login.html.php -->`

File: security/form_login.rst [hunk 7]
```diff
@@ -218,12 +246,19 @@ this by setting ``use_referer`` to true (it defaults to false):
             ),
         ));
 
-Redirecting on Login Failure
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. note::
+
+    The referrer URL is only used when is different from the URL generated by
```
Comment (xabbuh) [inline security/form_login.rst]: [...] when it is [...]

File: security/form_login.rst [hunk 8]
```diff
@@ -218,12 +246,19 @@ this by setting ``use_referer`` to true (it defaults to false):
...
+    the ``login_path`` route, to avoid a redirection loop.
```
Comment (xabbuh) [inline security/form_login.rst]: the comma should be removed

File: security/form_login.rst [hunk 9]
```diff
@@ -271,65 +306,46 @@ option to define the route or URL the user is redirected to:
...
-and ``_failure_path`` for login errors:
+.. code-block:: text
+
+    http://example.com/some/path?_failure_path=/forgot-password
 
 .. configuration-block::
 
     .. code-block:: html+twig
 
         {# src/AppBundle/Resources/views/Security/login.html.twig #}
```
Comment (xabbuh) [inline security/form_login.rst]: ``{# app/Resources/views/security/login.html.twig #}`

File: security/form_login.rst [hunk 10]
```diff
@@ -271,65 +306,46 @@ option to define the route or URL the user is redirected to:
...
-            <input type="text" id="username" name="_username" value="{{ last_username }}" />
-
-            <label for="password">Password:</label>
-            <input type="password" id="password" name="_password" />
-
-            <input type="hidden" name="_target_path" value="account" />
-            <input type="hidden" name="_failure_path" value="login" />
+            {# ... #}
 
+            <input type="hidden" name="_failure_path" value="{{ path('forgot-password') }}" />
```
Comment (xabbuh) [inline security/form_login.rst]: `forgot_password`

File: security/form_login.rst [hunk 11]
```diff
@@ -271,65 +306,46 @@ option to define the route or URL the user is redirected to:
...
-            <input type="text" id="username" name="_username" value="<?php echo $last_username ?>" />
-
-            <label for="password">Password:</label>
-            <input type="password" id="password" name="_password" />
-
-            <input type="hidden" name="_target_path" value="account" />
-            <input type="hidden" name="_failure_path" value="login" />
+            // ...
 
+            <input type="hidden" name="_failure_path" value="<?php echo $view['router']->generate('forgot-password') ?>" />
```
Comment (xabbuh) [inline security/form_login.rst]: `forgot_password`

File: security/form_login.rst [hunk 12]
```diff
@@ -271,65 +306,46 @@ option to define the route or URL the user is redirected to:
...
         <form action="<?php echo $view['router']->generate('login') ?>" method="post">
-            <label for="username">Username:</label>
-            <input type="text" id="username" name="_username" value="<?php echo $last_username ?>" />
-
-            <label for="password">Password:</label>
-            <input type="password" id="password" name="_password" />
-
-            <input type="hidden" name="_target_path" value="account" />
-            <input type="hidden" name="_failure_path" value="login" />
+            // ...
```
Comment (xabbuh) [inline security/form_login.rst]: `<!-- ... -->`

File: security/form_login.rst [hunk 13]
```diff
@@ -271,65 +306,46 @@ option to define the route or URL the user is redirected to:
...
-            <input type="hidden" name="_failure_path" value="login" />
+            {# ... #}
 
+            <input type="hidden" name="_failure_path" value="{{ path('forgot-password') }}" />
             <input type="submit" name="login" />
         </form>
 
     .. code-block:: html+php
 
         <!-- src/AppBundle/Resources/views/Security/login.html.php -->
```
Comment (xabbuh) [inline security/form_login.rst]: `<!-- app/Resources/views/security/login.html.php -->`

File: security/form_login.rst [hunk 14]
```diff
@@ -377,9 +393,40 @@ The name of the hidden fields in the login form is also configurable using the
...
+
+.. code-block:: text
+
+    http://example.com/some/path?go_to=/dashboard&back_to=/forgot-password
+
+.. configuration-block::
+
+    .. code-block:: html+twig
+
+        {# src/AppBundle/Resources/views/Security/login.html.twig #}
```
Comment (xabbuh) [inline security/form_login.rst]: `{# app/Resources/views/security/login.html.twig #}`

File: security/form_login.rst [hunk 15]
```diff
@@ -377,9 +393,40 @@ The name of the hidden fields in the login form is also configurable using the
...
+            <input type="hidden" name="go_to" value="{{ path('dashboard') }}" />
+            <input type="hidden" name="back_to" value="{{ path('forgot-password') }}" />
+            <input type="submit" name="login" />
+        </form>
+
+    .. code-block:: html+php
+
+        <!-- src/AppBundle/Resources/views/Security/login.html.php -->
+        <form action="<?php echo $view['router']->generate('login') ?>" method="post">
+            // ...
```
Comment (xabbuh) [inline security/form_login.rst]: `<!-- ... -->`

File: security/form_login.rst [hunk 16]
```diff
@@ -377,9 +393,40 @@ The name of the hidden fields in the login form is also configurable using the
...
+            {# ... #}
+
+            <input type="hidden" name="go_to" value="{{ path('dashboard') }}" />
+            <input type="hidden" name="back_to" value="{{ path('forgot-password') }}" />
+            <input type="submit" name="login" />
+        </form>
+
+    .. code-block:: html+php
+
+        <!-- src/AppBundle/Resources/views/Security/login.html.php -->
```
Comment (xabbuh) [inline security/form_login.rst]: `<!-- app/Resources/views/security/login.html.php -->`

File: security/form_login.rst [hunk 17]
```diff
@@ -377,9 +393,40 @@ The name of the hidden fields in the login form is also configurable using the
...
+.. configuration-block::
+
+    .. code-block:: html+twig
+
+        {# src/AppBundle/Resources/views/Security/login.html.twig #}
+        <form action="{{ path('login') }}" method="post">
+            {# ... #}
+
+            <input type="hidden" name="go_to" value="{{ path('dashboard') }}" />
+            <input type="hidden" name="back_to" value="{{ path('forgot-password') }}" />
```
Comment (xabbuh) [inline security/form_login.rst]: `forgot_password`

File: security/form_login.rst [hunk 18]
```diff
@@ -377,9 +393,40 @@ The name of the hidden fields in the login form is also configurable using the
...
+        </form>
+
+    .. code-block:: html+php
+
+        <!-- src/AppBundle/Resources/views/Security/login.html.php -->
+        <form action="<?php echo $view['router']->generate('login') ?>" method="post">
+            // ...
+
+            <input type="hidden" name="go_to" value="<?php echo $view['router']->generate('dashboard') ?>" />
+            <input type="hidden" name="back_to" value="<?php echo $view['router']->generate('forgot-password') ?>" />
```
Comment (xabbuh) [inline security/form_login.rst]: `forgot_password`

File: security/form_login.rst (hunk 4 above)
Comment (xabbuh) [inline security/form_login.rst]: the leading whitespace must be removed by the way (see the build failure)

**Comment by javiereguiluz** [comment]
@xabbuh thank you for the time you dedicated to review this. As usual, a great review! Thanks.

**Comment by xabbuh** [comment]
Thank you Javier.

**Pull request closed without merging**